proxy/ProxyTransaction.cc

```diff
@@ -220,3 +220,78 @@ ProxyTransaction::has_request_body(int64_t request_content_length, bool is_chunk
 {
   return request_content_length > 0 || is_chunked;
 }
+
+bool
+ProxyTransaction::is_read_closed() const
+{
+  return false;
+}
+
+bool
+ProxyTransaction::expect_send_trailer() const
+{
+  return false;
+}
+
+void
+ProxyTransaction::set_expect_send_trailer()
+{
+}
+bool
+ProxyTransaction::expect_receive_trailer() const
+{
+  return false;
+}
+
+void
+ProxyTransaction::set_expect_receive_trailer()
+{
+}
+
+void
+ProxyTransaction::attach_transaction(HttpSM *attach_sm)
+{
+  _sm = attach_sm;
+}
+
+void
+ProxyTransaction::release()
+{
+  HttpTxnDebug("[%" PRId64 "] session released by sm [%" PRId64 "]", _proxy_ssn ? _proxy_ssn->connection_id() : 0,
+               _sm ? _sm->sm_id : 0);
+
+  this->decrement_transactions_stat();
```

Comment:
Interesting.  The declaration has this set to 0.  The Http2Stream was calling the super version explicitly.  Http1 wasn't calling it at all.  I'm eliminating it in the ProxyTransaction and just inlining the necessary bits into the Http2Stream use cases.

Need to think some more if there are double decrements coming from transaction_done.  Probably best to do the decrements in the shutdown logic instead of transaction_done.   Need to do another round of production testing.

Reply:
Looking at my current prod run, we are double decrementing.  My current_transaction metrics are all going to 0.  Will track this down.

Reply:
I think the double decrement has been dealt with.  The current_*_connection, current_*_transaction and current_*_stream counts seemed to be staying accurate in my production run earlier today.